internal/registry/json_test.go

```diff
@@ -0,0 +1,343 @@
+
+	require.Equal(t, imageManifestV1{
+		Config: descriptorV1{
+			MediaType: "application/vnd.docker.container.image.v1+json",
+			Digest:    "sha256:00378fa4979bfcc7d1f5d33bb8cebe526395021801f9e233f8909ffc25a6f630",
+			Size:      3744,
+		},
+		Layers: []descriptorV1{
+			{"application/vnd.docker.image.rootfs.foreign.diff.tar.gzip", "sha256:4612f6d0b889cad0ed0292fae3a0b0c8a9e49aff6dea8eb049b2386d9b07986f", 1718332879},
+			{"application/vnd.docker.image.rootfs.foreign.diff.tar.gzip", "sha256:399f118dfaa9a753e98d128238b944432c7bcabea88a2998a6efbbece28ed303", 751421005},
+			{mediaTypeDockerLayer, "sha256:47916aee02007e0e175e80deb2938cf8f95457b9abb555bd44dc461680dc552c", 323887},
+			{mediaTypeDockerLayer, "sha256:ba79ee4428b5ceec3026664126a146fd8c1041b478f3018ec0c90b78d7fe6355", 331919},
+			{mediaTypeDockerLayer, "sha256:fd103a6c37aad8ffeaef6521612ed5a5153b104fffdb8bf3b6cf3d0beaaa49c4", 12217107},
+			{mediaTypeDockerLayer, "sha256:0fcfdc906e922391139a1c2d8f5d600066fa3b21c720a4024831471e2a8f0011", 337530},
+			{mediaTypeDockerLayer, "sha256:f5ece8fbad694f5d1169c17ddd4217265cdf3dd886b71a8e9144f8b00e22de07", 2410},
+			{mediaTypeDockerLayer, "sha256:8d3db7768af4371ec3f749f6816c8450687e276a883b8ca626a1fc1402fd32e0", 419457},
+			{mediaTypeDockerLayer, "sha256:f0b13e108f65feef6ee7b28a639a516aa37082bca3e0ac332bcde1e97e095b6b", 1303},
+			{mediaTypeDockerLayer, "sha256:9e17bb8cfb82c53b1793341a2dfb555e63088b1594d81d2b01106fae9a8aa60b", 1745},
+			{mediaTypeDockerLayer, "sha256:30188a58a9ae8bd6cbfc36a6ba873a1a8cfe5a50993fc982844b935fa2724126", 1327},
+			{mediaTypeDockerLayer, "sha256:ce93263143f489be1ca45bbda23e98dc97445fc9b3d53a7ffd4f7a7eb25889fc", 1334},
+		},
+	}, v)
+}
+
+func TestNewImage_Windows(t *testing.T) {
+	var i imageManifestV1
+	require.NoError(t, json.Unmarshal(windowsVndDockerImageManifestV1Json, &i))
+	var c imageConfigV1
+	require.NoError(t, json.Unmarshal(windowsVndDockerImageConfigV1Json, &c))
+
+	require.Equal(t, &internal.Image{
+		Platform: internal.OSWindows + "/" + internal.ArchAmd64,
+		FilesystemLayers: []*internal.FilesystemLayer{
+			{
+				URL:       "/blobs/sha256:47916aee02007e0e175e80deb2938cf8f95457b9abb555bd44dc461680dc552c",
+				MediaType: "application/vnd.docker.image.rootfs.diff.tar.gzip",
+				Size:      323887,
+				CreatedBy: `cmd /S /C mkdir "C:\\Program\ Files\\envoy"`,
+			},
+			{
+				URL:       "/blobs/sha256:ba79ee4428b5ceec3026664126a146fd8c1041b478f3018ec0c90b78d7fe6355",
+				MediaType: "application/vnd.docker.image.rootfs.diff.tar.gzip",
+				Size:      331919,
+				CreatedBy: `cmd /S /C setx path "%path%;c:\Program Files\envoy"`,
+			},
+			{
+				URL:       "/blobs/sha256:fd103a6c37aad8ffeaef6521612ed5a5153b104fffdb8bf3b6cf3d0beaaa49c4",
+				MediaType: "application/vnd.docker.image.rootfs.diff.tar.gzip",
+				Size:      12217107,
+				CreatedBy: `cmd /S /C #(nop) ADD file:61df7bfb8255c0673d4ed25f961df5121141ee800202081e549fc36828624577 in C:\Program Files\envoy\ `,
```

Comment:
Ex. this is the actual layer that includes the envoy binary. getting this data in any existing tool is really hard because this image is in windows format which has a lot of implications not the least of which needing to match the host OS https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility?tabs=windows-server-20H2%2Cwindows-10-20H2